ansible/roles/test/files/acstests/IP_decap_test.py

```diff
@@ -0,0 +1,193 @@
+Parameters:    route_info - The route_info file location 
+                       lo_ip -  The loop_back IP that is configured in the decap rule
+                       router_mac - The mac of the router_mac
+                       dscp_mode - The rule for the dscp parameter in the decap packet that is configured in the JSON file ('pipe' for inner and 'uniform' for outer)
+                
+'''
+
+#---------------------------------------------------------------------
+# Global imports
+#---------------------------------------------------------------------
+import random
+import time
+import logging
+import ptf.packet as scapy
+import socket
+import ptf.dataplane as dataplane
+
+from ptf.testutils import *
+from ptf.mask import Mask
+import ipaddress
+
+import os
+import unittest
+
+import ptf
+from ptf.base_tests import BaseTest
+from ptf import config
+import ptf.dataplane as dataplane
+import ptf.testutils as testutils
+
+import pprint
+from router_utils import *
+
+
+#---------------------------------------------------------------------
+# Global variables
+#---------------------------------------------------------------------
+PREFIX_AND_PORT_SPLITTER=" "
+PORT_LIST_SPLITTER=","
+PORT_COUNT = 31
+
+class DecapPacketTest(BaseTest, RouterUtility):
+    def __init__(self):
+        BaseTest.__init__(self)
+        self.test_params = testutils.test_params_get()
+	#-----------------------------------------------------------------
+    def setUp(self):
+        '''
+        @summary: Setup for the test
+        '''
+        self.dataplane = ptf.dataplane_instance
+        self.router_mac = self.test_params['router_mac']
+    #-----------------------------------------------------------------
+    
+    def send_and_verify(self, dst_ip, expected_ports, src_port, triple_encap = False):
+        '''
+        @summary: This function builds encap packet, send and verify their arrival, When a packet will not arrived as expected an exeption will be throwen 
+        @dst_ip: the destination ip for the inner IP header
+        @expected_ports: list of ports that a packet can arrived from 
+        @src_port: the physical port that the packet will be sent from 
+        @triple_encap: Bool if to send triple encapsulated packet
+        '''
+        #setting parameters
+        src_mac =  self.dataplane.get_mac(0, 0)
+        dst_mac = '00:11:22:33:44:55'
+        outer_src_ip = '1.1.1.1'
+        inner_src_ip = '2.2.2.2'
+        router_mac = self.test_params['router_mac']
+        dscp_in = random.randint(0, 32)
+        tos_in = dscp_in << 2
+        dscp_out = random.randint(0, 32)
+        tos_out = dscp_out << 2
+        outer_ttl = random.randint(2, 63) 
+        sport=1234
+        dport=80
+        if ("pipe" == self.test_params['dscp_mode']):
```

Comment:
some fields only filled once as some parameters only for the inner (end with *_in) and some for only the outer
in addition, simple_ipv4ip_packet there isn't all of the fields like simple_tcp_packet
its only another layer of L2 and L3 

But in simple_tcp_packet I added tcp ports for more detailed code
Because it's the default values, so it will not change anything